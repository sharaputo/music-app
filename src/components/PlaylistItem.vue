<template>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  >
    <div>
      <RouterLink
        :to="{ name: 'track', params: { id: track.docId } }"
        class="font-bold block text-gray-600"
      >
        {{ track.modified_name }}
      </RouterLink>
      <p class="text-gray-500 text-sm first-letter:uppercase">
        {{ track.display_name }}
      </p>
    </div>

    <div class="text-gray-600 text-lg">
      <RouterLink
        :to="{ name: 'track', params: { id: track.docId }, hash: '#comments' }"
        custom
        v-slot="{ navigate }"
      >
        <span class="comments" @click="navigate">
          {{ track.comment_count }}
          <span v-icon:full="'fa fa-comments text-gray-600'"></span>
        </span>
      </RouterLink>
    </div>
  </li>
</template>

<script setup lang="ts">
import type { TrackDetails } from "@/types/TrackDetails";

defineProps<{
  track: TrackDetails;
}>();
</script>
