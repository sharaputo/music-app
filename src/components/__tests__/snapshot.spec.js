import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import PlaylistItem from "@/components/PlaylistItem.vue";

describe("Snapshot PlaylistItem.vue", () => {
  it("renders correctly", () => {
    const track = {
      docId: "123",
      modified_name: "test",
      display_name: "test 2",
      comment_count: 2,
    };

    const wrapper = shallowMount(PlaylistItem, {
      propsData: { track },
      global: {
        stubs: { RouterLink: RouterLinkStub },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
