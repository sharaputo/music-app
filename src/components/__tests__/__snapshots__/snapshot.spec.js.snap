// Vitest Snapshot v1

exports[`Snapshot PlaylistItem.vue > renders correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
>
  <div>
    <a
      class="font-bold block text-gray-600"
    >
      test
    </a>
    <p
      class="text-gray-500 text-sm first-letter:uppercase"
    >
      test 2
    </p>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    
    <span
      class="comments"
    >
      2 
      <span />
    </span>
    
  </div>
</li>
`;
